### PR TITLE
[JB-2499] Improve watcher speed by using api to reload apps

### PR DESCRIPTION
**Pull request by chrisgemignani**
Ticket: https://juiceanalytics.atlassian.net/browse/JB-2499
Type: Improvement

#### This PR introduces the following changes

- Add `jbapiutil` which can get an admin token and use it to load an app

## Documentation

This PR creates a new approach for loading apps that is much quicker than calling the django loadjuiceboxapp management command within the container.

### Approach

This approach assumes the default admin user exists, gets a token representing that admin user and uses that token to call LoadAppView.

### Performance

Time to load a sample app is 2-3 seconds (using api) vs 14 seconds (via management command).
```
# using api
$ time jb add jbodemo_birdo . 
jb add jbodemo_birdo  0.55s user 0.43s system 43% cpu 2.233 total

# using management command
$ time jb add jbodemo_birdo . 
jb add jbodemo_birdo  0.79s user 0.59s system 9% cpu 14.519 total
```

### Results

Results look like this

```
$ jb add jbodemo_birdo
Checking to see if Juicebox is running...
App jbodemo_birdo already exists.
Got admin token from storage
jbodemo_birdo was added successfully via API.jbodemo_birdo was added successfully via API.
[info]    Making image background

               image: logos/juicebox.png
               theme: /code/fruition/settings/../../apps/jbodemo_birdo/theme/theme.yaml

[info]    Syncing file: File is already uploaded to s3

                file: /code/fruition/settings/../../apps/jbodemo_birdo/theme/img/logos/juicebox.png
              s3_key: app_assets/de726b3b/img/logos/juicebox.png
               theme: /code/fruition/settings/../../apps/jbodemo_birdo/theme/theme.yaml

[info]    Syncing file: File is already uploaded to s3

                file: /code/apps/jbodemo_birdo/theme/img/bright-squares.png
              s3_key: app_assets/de726b3b/img/bright-squares.png
               theme: /code/fruition/settings/../../apps/jbodemo_birdo/theme/theme.yaml

```

Would also be really nice to pair w an autoreloader. 

## Checklist

- [ ] Add information to the release notes documentation
- [ ] Add new feature to the usage documentation
- [x] Add tests

### CODE REVIEW

File: jbcli/jbcli/utils/jbapiutil.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from requests import post
+import os
+from .storageutil import stash
+from .format import *
+
```
Comment (chrisgemignani) [inline jbcli/jbcli/utils/jbapiutil.py]: Not very keen on this 👇  but don't have a better idea.

File: jbcli/jbcli/cli/jb.py [hunk 2]
```diff
@@ -135,10 +135,11 @@ def add(applications, add_desktop):
                                 'Failed to add {} to Github Desktop.'.format(
                                     app))
                 try:
-                    dockerutil.run(
-                        '/venv/bin/python manage.py loadjuiceboxapp {}'.format(
-                            app))
-                    echo_success('{} was added successfully.'.format(app))
+                    if not jbapiutil.load_app(app):
```
Comment (stevenwally) [inline jbcli/jbcli/cli/jb.py]: Do we also want to utilize this for the `watch` command?

Comment (stevenwally) [inline jbcli/jbcli/cli/jb.py]: Can we also utilize this here? https://github.com/juiceinc/devlandia/blob/3937becc0a3e61b649ad0406878c50d9dae57c34/jbcli/jbcli/utils/dockerutil.py#L36

Comment (chrisgemignani) [reply jbcli/jbcli/cli/jb.py]: Yup, I wanted to wait until we got yours merged.

**Comment by stevenwally** [comment]
Most recent changes look good. I'm cool with going a head with merging this.

**Pull request merged**